dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Guest/AddSimpleProductToCartTest.php

```diff
@@ -79,6 +80,54 @@ public function testAddSimpleProductToCart()
         self::assertEquals('USD', $rowTotalIncludingTax['currency']);
     }
 
+    /**
+     * Add disabled product to cart
+     *
+     * @magentoApiDataFixture Magento/Catalog/_files/multiple_products.php
+     * @magentoApiDataFixture Magento/GraphQl/Quote/_files/guest/create_empty_cart.php
+     * @return void
+     */
+    public function testAddDisabledProductToCart(): void
+    {
+        $sku = 'simple3';
+        $quantity = 2;
+
+        $maskedQuoteId = $this->getMaskedQuoteIdByReservedOrderId->execute('test_quote');
+        $query = $this->getQuery($maskedQuoteId, $sku, $quantity);
+
+        $this->expectException(ResponseContainsErrorsException::class);
+        $this->expectExceptionMessage(
+            'Could not add the product with SKU ' . $sku . ' to the shopping cart: ' .
+            'Product that you are trying to add is not available.'
+        );
+
+        $this->graphQlMutation($query);
+    }
+
+    /**
+     * Add out of stock product to cart
+     *
+     * @magentoApiDataFixture Magento/Catalog/_files/product_virtual_out_of_stock.php
+     * @magentoApiDataFixture Magento/GraphQl/Quote/_files/guest/create_empty_cart.php
+     * @return void
+     */
+    public function testAddOutOfStockProductToCart(): void
```

Comment:
For virtual product we have separate mutation https://devdocs.magento.com/guides/v2.3/graphql/mutations/add-virtual-products.html

Please change the test to make following:
1. Add `simple` (should be in stock) - using fixture
2. Add another one `simple` - using fixture
3. Mark one of them out of stock - using fixture - `Magento/GraphQl/Catalog/_files/set_simple_product_out_of_stock.php`
4. Perform mutation

See good example here: `Magento\GraphQl\Quote\Guest\PlaceOrderTest::testPlaceOrderWithOutOfStockProduct`

Reply:
Yes, I've covered this case. Please check that